packages/app-builder-lib/src/node-module-collector/types.ts

```diff
@@ -11,6 +11,7 @@ export interface DependencyTree {
   readonly from?: string
   readonly workspaces?: string[]
   readonly path: string
+  skipCircularDeps?: boolean
```

Comment:
I'm wary of using a boolean that is optional and only ever set true in the code logic flow.

Maybe we can use `private circularDependencyDetected = false`?